src/menu/MenuBar.ts

```diff
@@ -0,0 +1,105 @@
+import { createHandle } from '@dojo/core/lang';
+import { debounce } from '@dojo/core/util';
+import { v, w } from '@dojo/widget-core/d';
+import { DNode } from '@dojo/widget-core/interfaces';
+import ThemeableMixin, { theme, ThemeableProperties } from '@dojo/widget-core/mixins/Themeable';
+import WidgetBase from '@dojo/widget-core/WidgetBase';
+import * as css from './styles/menuBar.m.css';
+import SlidePane from '../slidepane/SlidePane';
+import { observeViewport } from '../common/util';
```

Comment:
Why does the widget need to observe the viewport? The viewport should be passed via properties.